src/GraphQL/Resolver/ContentThumbnailResolver.php

```diff
@@ -6,88 +6,40 @@
  */
 namespace EzSystems\EzPlatformGraphQL\GraphQL\Resolver;
 
-use Exception;
 use eZ\Publish\API\Repository\Values\Content\Content;
-use eZ\Publish\API\Repository\Values\Content\Field;
-use eZ\Publish\API\Repository\Values\Content\Query\Criterion;
-use eZ\Publish\Core\FieldType;
-use eZ\Publish\SPI\Variation\VariationHandler;
-use EzSystems\EzPlatformGraphQL\GraphQL\DataLoader\ContentLoader;
+use eZ\Publish\SPI\Repository\Strategy\ContentThumbnail\ThumbnailStrategy;
 
 final class ContentThumbnailResolver
 {
-    const THUMBNAIL_VARIATION_IDENTIFIER = 'medium';
-
-    /**
-     * @var \EzSystems\EzPlatformGraphQL\GraphQL\DataLoader\ContentLoader
-     */
-    private $contentLoader;
-    /**
-     * @var \eZ\Publish\Core\FieldType\ImageAsset\AssetMapper
-     */
-    private $assetMapper;
-    /**
-     * @var \eZ\Publish\SPI\Variation\VariationHandler
-     */
-    private $variationHandler;
-    /**
-     * @var \eZ\Publish\Core\FieldType\Image\Type
-     */
-    private $imageFieldType;
+    /** @var \eZ\Publish\SPI\Repository\Strategy\ContentThumbnail\ThumbnailStrategy */
+    private $thumbnailStrategy;
 
     public function __construct(
-        FieldType\Image\Type $imageFieldType,
-        ContentLoader $contentLoader,
-        FieldType\ImageAsset\AssetMapper $assetMapper,
-        VariationHandler $variationHandler)
-    {
-        $this->contentLoader = $contentLoader;
-        $this->assetMapper = $assetMapper;
-        $this->variationHandler = $variationHandler;
-        $this->imageFieldType = $imageFieldType;
+        ThumbnailStrategy $thumbnailStrategy
+    ) {
+        $this->thumbnailStrategy = $thumbnailStrategy;
     }
 
     /**
      * @return array|null array with the thumbnail info, or null if no thumbnail could be obtained for that image
      */
     public function resolveContentThumbnail(Content $content): ?array
     {
-        try {
-            $imageField = $this->getThumbnailImageField($content);
-        } catch (Exception $e) {
-            return null;
-        }
+        $thumbnail = $this->thumbnailStrategy->getThumbnail(
+            $content->getContentType(),
+            $content->getFields(),
+            $content->getVersionInfo()
+        );
 
-        if ($imageField === null || $this->imageFieldType->isEmptyValue($imageField->value)) {
+        if ($thumbnail === null) {
             return null;
         }
 
-        $thumbnailVariation = $this->variationHandler->getVariation($imageField, $content->versionInfo, self::THUMBNAIL_VARIATION_IDENTIFIER);
-
         return [
-            'uri' => $thumbnailVariation->uri,
-            'width' => $thumbnailVariation->width,
-            'height' => $thumbnailVariation->height,
-            'alternativeText' => $imageField->value->alternativeText,
+            'uri' => $thumbnail->resource,
+            'width' => $thumbnail->width,
+            'height' => $thumbnail->height,
+            'mimeType' => $thumbnail->mimeType,
```

Comment:
Missing `alternativeText` in this array might cause issues. Is this intentional?

Reply:
>intentional
yes. Our thumbnail object (\eZ\Publish\API\Repository\Values\Content\Thumbnail) does not have alternative text atm, it would have to be implemented for every `\eZ\Publish\SPI\Repository\Strategy\ContentThumbnail\Field\ThumbnailStrategy` we have.

On the other hand, you maybe right and till it will happen maybe we should just put empty string there.